nf_requirements/v1alpha1/interface_types.go

```diff
@@ -16,21 +16,43 @@ limitations under the License.
 
 package v1alpha1
 
-type AttachmentType string
-
-// AttachmentTypeNone defines an untagged attachement (no VLAN)
-const AttachmentTypeNone AttachmentType = "none"
-
-// AttachmentTypeVLAN defines a tagged/vlan attachement
-const AttachmentTypeVLAN AttachmentType = "vlan"
-
-type CNIType string
-
-// CNITypeSRIOV defines the sriov cni
-const CNITypeSRIOV CNIType = "sriov"
-
-// CNITypeIPVLAN defines the ipvlan cni
-const CNITypeIPVLAN CNIType = "ipvlan"
-
-// CNITypeMACVLAN defines the macvlan cni
-const CNITypeMACVLAN CNIType = "macvlan"
+import (
+	"reflect"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+)
+
+// +kubebuilder:object:root=true
+type Interface struct {
+	metav1.TypeMeta   `json:",inline" yaml:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" yaml:"metadata,omitempty"`
+
+	Spec   InterfaceSpec   `json:"spec,omitempty" yaml:"spec,omitempty"`
+	Status InterfaceStatus `json:"status,omitempty" yaml:"status,omitempty"`
+}
+
+// TBD how do we distinguish the loopback from the vnic(s)
+type InterfaceSpec struct {
+	// NetworkInstance defines the networkInstance to which this interface belongs
+	// Name and optionally Namespace is used here
+	NetworkInstance *corev1.ObjectReference `json:"networkInstance" yaml:"networkInstance"`
+	// CNIType defines the cniType that is used to attach the interface to the pod
+	// +kubebuilder:validation:Enum=sriov;ipvlan;macvlan
+	CNIType CNIType `json:"cniType,omitempty" yaml:"cniType,omitempty"`
+	// AttachmentType defines if the interface is attached using a vlan or not
+	// +kubebuilder:validation:Enum=none;vlan
+	AttachmentType AttachmentType `json:"attachmentType,omitempty" yaml:"attachmentType,omitempty"`
+}
+
+type InterfaceStatus struct {
```

Comment:
curious about why the struct is empty? Thanks

Reply:
so far we had no reason to add something in the status and it could be empty. Now with the new fn-runtime or sdk proposal we will add information here going fwd. So it will be used to indicate the IP(s) that were allocated e.g.
As such the nf-deploy function will not have to look to intermediate CR(s) to understand the IP.